### :x: 13 Tests Failed:
| Tests completed | Failed | Passed | Skipped |
|---|---|---|---|
| 2219 | 13 | 2206 | 194 |
<details><summary>View the top 3 failed test(s) by shortest run time</summary>

> <pre><code class="language-python">test.unit.test_setup_minicore_pruner::test_native_subdir_matches_platform[AIX-ppc64-libc7-aix_ppc64]</code></pre>
> <details><summary>Stack Traces | 0.001s run time</summary>
> 
> > <pre><code class="language-python">test/unit/test_setup_minicore_pruner.py:42: in setup_module
> >     return _load_setup_module()
> > test/unit/test_setup_minicore_pruner.py:36: in _load_setup_module
> >     spec.loader.exec_module(module)
> > &lt;frozen importlib._bootstrap_external&gt;:759: in exec_module
> >     ???
> > &lt;frozen importlib._bootstrap&gt;:491: in _call_with_frames_removed
> >     ???
> > setup.py:9: in &lt;module&gt;
> >     from setuptools import Extension, setup
> > E   ModuleNotFoundError: No module named &#x27;setuptools&#x27;</code></pre>
> 
> </details>


> <pre><code class="language-python">test.unit.test_setup_minicore_pruner::test_native_subdir_matches_platform[Darwin-arm64-libc4-macos_aarch64]</code></pre>
> <details><summary>Stack Traces | 0.001s run time</summary>
> 
> > <pre><code class="language-python">test/unit/test_setup_minicore_pruner.py:42: in setup_module
> >     return _load_setup_module()
> > test/unit/test_setup_minicore_pruner.py:36: in _load_setup_module
> >     spec.loader.exec_module(module)
> > &lt;frozen importlib._bootstrap_external&gt;:759: in exec_module
> >     ???
> > &lt;frozen importlib._bootstrap&gt;:491: in _call_with_frames_removed
> >     ???
> > setup.py:9: in &lt;module&gt;
> >     from setuptools import Extension, setup
> > E   ModuleNotFoundError: No module named &#x27;setuptools&#x27;</code></pre>
> 
> </details>


> <pre><code class="language-python">test.unit.test_setup_minicore_pruner::test_native_subdir_matches_platform[Darwin-x86_64-libc5-macos_x86_64]</code></pre>
> <details><summary>Stack Traces | 0.001s run time</summary>
> 
> > <pre><code class="language-python">test/unit/test_setup_minicore_pruner.py:42: in setup_module
> >     return _load_setup_module()
> > test/unit/test_setup_minicore_pruner.py:36: in _load_setup_module
> >     spec.loader.exec_module(module)
> > &lt;frozen importlib._bootstrap_external&gt;:759: in exec_module
> >     ???
> > &lt;frozen importlib._bootstrap&gt;:491: in _call_with_frames_removed
> >     ???
> > setup.py:9: in &lt;module&gt;
> >     from setuptools import Extension, setup
> > E   ModuleNotFoundError: No module named &#x27;setuptools&#x27;</code></pre>
> 
> </details>


> <pre><code class="language-python">test.unit.test_setup_minicore_pruner::test_native_subdir_matches_platform[Linux-aarch64-libc1-linux_aarch64_glibc]</code></pre>
> <details><summary>Stack Traces | 0.001s run time</summary>
> 
> > <pre><code class="language-python">test/unit/test_setup_minicore_pruner.py:42: in setup_module
> >     return _load_setup_module()
> > test/unit/test_setup_minicore_pruner.py:36: in _load_setup_module
> >     spec.loader.exec_module(module)
> > &lt;frozen importlib._bootstrap_external&gt;:759: in exec_module
> >     ???
> > &lt;frozen importlib._bootstrap&gt;:491: in _call_with_frames_removed
> >     ???
> > setup.py:9: in &lt;module&gt;
> >     from setuptools import Extension, setup
> > E   ModuleNotFoundError: No module named &#x27;setuptools&#x27;</code></pre>
> 
> </details>


> <pre><code class="language-python">test.unit.test_setup_minicore_pruner::test_native_subdir_matches_platform[Linux-aarch64-libc3-linux_aarch64_musl]</code></pre>
> <details><summary>Stack Traces | 0.001s run time</summary>
> 
> > <pre><code class="language-python">test/unit/test_setup_minicore_pruner.py:42: in setup_module
> >     return _load_setup_module()
> > test/unit/test_setup_minicore_pruner.py:36: in _load_setup_module
> >     spec.loader.exec_module(module)
> > &lt;frozen importlib._bootstrap_external&gt;:759: in exec_module
> >     ???
> > &lt;frozen importlib._bootstrap&gt;:491: in _call_with_frames_removed
> >     ???
> > setup.py:9: in &lt;module&gt;
> >     from setuptools import Extension, setup
> > E   ModuleNotFoundError: No module named &#x27;setuptools&#x27;</code></pre>
> 
> </details>


> <pre><code class="language-python">test.unit.test_setup_minicore_pruner::test_native_subdir_matches_platform[Linux-x86_64-libc2-linux_x86_64_musl]</code></pre>
> <details><summary>Stack Traces | 0.001s run time</summary>
> 
> > <pre><code class="language-python">test/unit/test_setup_minicore_pruner.py:42: in setup_module
> >     return _load_setup_module()
> > test/unit/test_setup_minicore_pruner.py:36: in _load_setup_module
> >     spec.loader.exec_module(module)
> > &lt;frozen importlib._bootstrap_external&gt;:759: in exec_module
> >     ???
> > &lt;frozen importlib._bootstrap&gt;:491: in _call_with_frames_removed
> >     ???
> > setup.py:9: in &lt;module&gt;
> >     from setuptools import Extension, setup
> > E   ModuleNotFoundError: No module named &#x27;setuptools&#x27;</code></pre>
> 
> </details>


> <pre><code class="language-python">test.unit.test_setup_minicore_pruner::test_native_subdir_matches_platform[Windows-AMD64-libc6-windows_x86_64]</code></pre>
> <details><summary>Stack Traces | 0.001s run time</summary>
> 
> > <pre><code class="language-python">test/unit/test_setup_minicore_pruner.py:42: in setup_module
> >     return _load_setup_module()
> > test/unit/test_setup_minicore_pruner.py:36: in _load_setup_module
> >     spec.loader.exec_module(module)
> > &lt;frozen importlib._bootstrap_external&gt;:759: in exec_module
> >     ???
> > &lt;frozen importlib._bootstrap&gt;:491: in _call_with_frames_removed
> >     ???
> > setup.py:9: in &lt;module&gt;
> >     from setuptools import Extension, setup
> > E   ModuleNotFoundError: No module named &#x27;setuptools&#x27;</code></pre>
> 
> </details>


> <pre><code class="language-python">test.unit.test_setup_minicore_pruner::test_native_subdir_returns_none_for_unknown_platform</code></pre>
> <details><summary>Stack Traces | 0.001s run time</summary>
> 
> > <pre><code class="language-python">test/unit/test_setup_minicore_pruner.py:42: in setup_module
> >     return _load_setup_module()
> > test/unit/test_setup_minicore_pruner.py:36: in _load_setup_module
> >     spec.loader.exec_module(module)
> > &lt;frozen importlib._bootstrap_external&gt;:759: in exec_module
> >     ???
> > &lt;frozen importlib._bootstrap&gt;:491: in _call_with_frames_removed
> >     ???
> > setup.py:9: in &lt;module&gt;
> >     from setuptools import Extension, setup
> > E   ModuleNotFoundError: No module named &#x27;setuptools&#x27;</code></pre>
> 
> </details>


> <pre><code class="language-python">test.unit.test_setup_minicore_pruner::test_prune_minicore_keeps_only_native_dir</code></pre>
> <details><summary>Stack Traces | 0.001s run time</summary>
> 
> > <pre><code class="language-python">test/unit/test_setup_minicore_pruner.py:42: in setup_module
> >     return _load_setup_module()
> > test/unit/test_setup_minicore_pruner.py:36: in _load_setup_module
> >     spec.loader.exec_module(module)
> > &lt;frozen importlib._bootstrap_external&gt;:759: in exec_module
> >     ???
> > &lt;frozen importlib._bootstrap&gt;:491: in _call_with_frames_removed
> >     ???
> > setup.py:9: in &lt;module&gt;
> >     from setuptools import Extension, setup
> > E   ModuleNotFoundError: No module named &#x27;setuptools&#x27;</code></pre>
> 
> </details>


> <pre><code class="language-python">test.unit.test_setup_minicore_pruner::test_prune_minicore_noop_when_directory_missing</code></pre>
> <details><summary>Stack Traces | 0.001s run time</summary>
> 
> > <pre><code class="language-python">test/unit/test_setup_minicore_pruner.py:42: in setup_module
> >     return _load_setup_module()
> > test/unit/test_setup_minicore_pruner.py:36: in _load_setup_module
> >     spec.loader.exec_module(module)
> > &lt;frozen importlib._bootstrap_external&gt;:759: in exec_module
> >     ???
> > &lt;frozen importlib._bootstrap&gt;:491: in _call_with_frames_removed
> >     ???
> > setup.py:9: in &lt;module&gt;
> >     from setuptools import Extension, setup
> > E   ModuleNotFoundError: No module named &#x27;setuptools&#x27;</code></pre>
> 
> </details>


> <pre><code class="language-python">test.unit.test_setup_minicore_pruner::test_prune_minicore_noop_when_platform_unknown</code></pre>
> <details><summary>Stack Traces | 0.001s run time</summary>
> 
> > <pre><code class="language-python">test/unit/test_setup_minicore_pruner.py:42: in setup_module
> >     return _load_setup_module()
> > test/unit/test_setup_minicore_pruner.py:36: in _load_setup_module
> >     spec.loader.exec_module(module)
> > &lt;frozen importlib._bootstrap_external&gt;:759: in exec_module
> >     ???
> > &lt;frozen importlib._bootstrap&gt;:491: in _call_with_frames_removed
> >     ???
> > setup.py:9: in &lt;module&gt;
> >     from setuptools import Extension, setup
> > E   ModuleNotFoundError: No module named &#x27;setuptools&#x27;</code></pre>
> 
> </details>


> <pre><code class="language-python">test.unit.test_setup_minicore_pruner::test_pruner_agrees_with_runtime_loader</code></pre>
> <details><summary>Stack Traces | 0.001s run time</summary>
> 
> > <pre><code class="language-python">test/unit/test_setup_minicore_pruner.py:42: in setup_module
> >     return _load_setup_module()
> > test/unit/test_setup_minicore_pruner.py:36: in _load_setup_module
> >     spec.loader.exec_module(module)
> > &lt;frozen importlib._bootstrap_external&gt;:759: in exec_module
> >     ???
> > &lt;frozen importlib._bootstrap&gt;:491: in _call_with_frames_removed
> >     ???
> > setup.py:9: in &lt;module&gt;
> >     from setuptools import Extension, setup
> > E   ModuleNotFoundError: No module named &#x27;setuptools&#x27;</code></pre>
> 
> </details>


> <pre><code class="language-python">test.unit.test_setup_minicore_pruner::test_native_subdir_matches_platform[Linux-x86_64-libc0-linux_x86_64_glibc]</code></pre>
> <details><summary>Stack Traces | 0.004s run time</summary>
> 
> > <pre><code class="language-python">test/unit/test_setup_minicore_pruner.py:42: in setup_module
> >     return _load_setup_module()
> > test/unit/test_setup_minicore_pruner.py:36: in _load_setup_module
> >     spec.loader.exec_module(module)
> > &lt;frozen importlib._bootstrap_external&gt;:759: in exec_module
> >     ???
> > &lt;frozen importlib._bootstrap&gt;:491: in _call_with_frames_removed
> >     ???
> > setup.py:9: in &lt;module&gt;
> >     from setuptools import Extension, setup
> > E   ModuleNotFoundError: No module named &#x27;setuptools&#x27;</code></pre>
> 
> </details>

</details>
<details><summary>View the full list of 1 :snowflake: flaky test(s)</summary>

> <pre><code class="language-python">test.unit.test_connection::test_handle_timeout[RETRY]</code></pre>
> **Flake rate in main:** 5.71% (Passed 33 times, Failed 2 times)
> <details><summary>Stack Traces | 9.59s run time</summary>
> 
> > <pre><code class="language-python">test/unit/test_connection.py:406: in test_handle_timeout
> >     assert 1 &lt; mockSessionRequest.call_count &lt; 4
> > E   AssertionError: assert 1 &lt; 1
> > E    +  where 1 = &lt;MagicMock name=&#x27;request&#x27; id=&#x27;5329896176&#x27;&gt;.call_count</code></pre>
> 
> </details>

</details>

To view more test analytics, go to the [Test Analytics Dashboard](https://app.codecov.io/gh/snowflakedb/snowflake-connector-python/tests/SNOW-3526469-prune-foreign-minicore-from-sdist)
<sub>📋 Got 3 mins? [Take this short survey](https://forms.gle/22i53Qa1CySZjA6c7) to help us improve Test Analytics.</sub>

--- a/test/unit/test_setup_minicore_pruner.py
+++ b/test/unit/test_setup_minicore_pruner.py
@@ -18,6 +18,11 @@ import pytest
 
 pytestmark = pytest.mark.skipolddriver
 
+pytest.importorskip(
+    "setuptools",
+    reason="setup.py imports setuptools at module scope; Python 3.14+ envs may not bundle it",
+)
+
 REPO_ROOT = Path(__file__).resolve().parents[2]
 SETUP_PY = REPO_ROOT / "setup.py"
 

